server/protocols/jmap/src/test/java/org/apache/james/jmap/model/NumberTest.java

```diff
@@ -0,0 +1,97 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.Test;
+
+import com.google.common.math.LongMath;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class NumberTest {
+    @Test
+    public void fromIntShouldReturnMinValueWhenNegativeValueWithLenient() throws Exception {
+        assertThat(Number.lenientFactory().fromInt(-1))
+            .isEqualTo(Number.ZERO);
+    }
+
+    @Test
+    public void fromLongShouldReturnMinValueWhenNegativeValueWithLenient() throws Exception {
+        assertThat(Number.lenientFactory().fromLong(-1))
+            .isEqualTo(Number.ZERO);
```

Comment:
at this point this is not lenient, this is normalization

Reply:
Doesn't all lenient parser apply normalization to not normal input in order to get a normal output?

I mean the normalization you apply in a lenient parser can be ignoring values/fields, sanitizing values, etc...